### PR TITLE
chore: module-wise configuration support

### PR DESCRIPTION
**Pull request by mrehan27**
helps: https://github.com/customerio/issues/issues/11648

### Changes

- Decoupled `ModuleTopLevelObject` from `SdkConfig`
- Moved `ModuleTopLevelObject` to `Common` in the process of decoupling feature modules from `Tracking`
- Removed dependencies on `SdkInitializedUtil` from modules
- Added `ModuleTopLevelConfig` to be used as a placeholder for config classes
- Added an empty `MessagingInAppConfigOptions` for the in-app module
- Commented code for for creating module instances, to be filled in by later PRs

### CODE REVIEW

File: Sources/Common/Module/ModuleTopLevelConfigOptions.swift [hunk 1]
```diff
@@ -0,0 +1,2 @@
+public protocol ModuleTopLevelConfig {
```
Comment (levibostian) [inline Sources/Common/Module/ModuleTopLevelConfigOptions.swift:1]: I currently don't see a use for this protocol in this PR. Perhaps it goes to a different PR?

File: Sources/MessagingInApp/MessagingInApp.swift [hunk 2]
```diff
@@ -21,8 +21,8 @@ public class MessagingInApp: ModuleTopLevelObject<MessagingInAppInstance>, Messa
     // constructor that is called by test classes
     // This function's job is to populate the `shared` property with
     // overrides such as DI graph.
-    override init(implementation: MessagingInAppInstance?, sdkInitializedUtil: SdkInitializedUtil) {
-        super.init(implementation: implementation, sdkInitializedUtil: sdkInitializedUtil)
+    override init(implementation: MessagingInAppInstance?) {
```
Comment (levibostian) [inline Sources/MessagingInApp/MessagingInApp.swift:24]: If I remember correctly, `ModuleTopLevelObject` was created to initialize modules in a way that worked with SDK initialization. By making push and in-app no longer tied to SDK initialization, I think that we can make `MessagingPush` and `MessagingInApp` no longer extend `ModuleTopLevelObject` and possibly even remove `ModuleTopLevelObject` entirely?

**Review by levibostian — APPROVED** [review]

File: Sources/Common/Module/ModuleTopLevelConfigOptions.swift (hunk 1 above)
Comment (mrehan27) [reply Sources/Common/Module/ModuleTopLevelConfigOptions.swift]: I was actually using this in upcoming PRs, but I no longer find this helpful, so removed this for now.

File: Sources/MessagingInApp/MessagingInApp.swift (hunk 2 above)
Comment (mrehan27) [reply Sources/MessagingInApp/MessagingInApp.swift:24]: Yes, we might be able to do that. But I'm a bit hesitant to remove this entirely at this point. I want to hold it till the feature modules are actually fully independent, we can then reevaluate this and remove if no longer needed. I have added a TODO so we don't forget this.

**Pull request merged**